devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,674 @@
+			tileAttribute("device.temperature", key: "PRIMARY_CONTROL") {
+				attributeState "temperature", label: '${currentValue}°',
+						backgroundColors:[
+								[value: 31, color: "#153591"],
+								[value: 44, color: "#1e9cbb"],
+								[value: 59, color: "#90d2a7"],
+								[value: 74, color: "#44b621"],
+								[value: 84, color: "#f1d801"],
+								[value: 95, color: "#d04e00"],
+								[value: 96, color: "#bc2323"]
+						]
+			}
+		}
+
+		valueTile("humidity", "device.humidity", inactiveLabel: false, width: 2, height: 2) {
+			state "humidity", label: '${currentValue}% humidity', unit: ""
+		}
+
+		valueTile("battery", "device.battery", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "battery", label: '${currentValue}% battery', unit: ""
+		}
+
+		valueTile("pendingChanges", "device.pendingChanges", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "pendingChanges", label:'${currentValue}'
+		}
+
+		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+			state "default", action: "refresh.refresh", icon: "st.secondary.refresh"
+		}
+
+		main(["contact", "temperature", "humidity"])
+		details(["contact", "temperature", "humidity", "battery", "refresh", "pendingChanges"])
+	}
+
+	// simulator metadata
+	simulator {
+		// status messages
+		status "open": "command: 2001, payload: FF"
+		status "closed": "command: 2001, payload: 00"
+		status "wake up": "command: 8407, payload: "
+	}
+
+	preferences {
+		configParams.each {
+			if (it.range) {
+				input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
+			} else {
+				input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options:it.options
+			}
+		}
+	}
+}
+
+def installed() {
+	logDebug "installed()..."
+	state.refreshConfig = true
+	sendEvent(name: "checkInterval", value: checkInterval, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+private static def getCheckInterval() {
+	// These are battery-powered devices, and it's not very critical
+	// to know whether they're online or not – 12 hrs
+	return (60 * 60 * 3) + (5 * 60)
+}
+
+def updated() {
+	if (!isDuplicateCommand(state.lastUpdated, 5000)) {
+		state.lastUpdated = new Date().time
+
+		logTrace "updated()"
+		if (device.latestValue("checkInterval") != checkInterval) {
+			sendEvent(name: "checkInterval", value: checkInterval, displayed: false)
+		}
+
+		refreshPendingChanges()
+
+		logForceWakeupMessage "Configuration changes will be sent to the device the next time it wakes up."
+	}
+}
+
+def configure() {
+	logTrace "configure()"
+
+	runIn(8, executeConfigure)
+}
+
+def executeConfigure() {
+	def cmds = [
+			sensorBinaryGetCmd(),
+			batteryGetCmd(),
+			sensorMultilevelGetCmd(tempSensorType),
```

Comment:
If You change minTemperatureOffsetParam or minHumidityOffsetParam - sensorMultilevelGetCmd will sent to the device twice because getCofingCmds will return that command too.

Reply:
Yes, this is my mistake, the order is repeated.